src/org/pentaho/reporting/platform/plugin/async/AsyncReportStatusListener.java

```diff
@@ -21,25 +21,31 @@
 import org.pentaho.reporting.engine.classic.core.event.ReportProgressEvent;
 import org.pentaho.reporting.engine.classic.core.event.ReportProgressListener;
 
-import java.util.Queue;
 import java.util.UUID;
-import java.util.concurrent.ConcurrentLinkedQueue;
 
 /**
  * Simple synchronized bean with async execution status.
- *
+ * <p/>
  * Created by dima.prokopenko@gmail.com on 2/12/2016.
  */
 class AsyncReportStatusListener implements IAsyncReportListener, ReportProgressListener, IAsyncReportState {
 
+  public static final String COMPUTING_LAYOUT = "async_computing_layout_title";
+  public static final String PRECOMPUTING_VALUES = "async_precomputing_values_title";
+  public static final String PAGINATING = "async_paginating_title";
+  public static final String GENERATING_CONTENT = "async_generating_content_title";
+
   private volatile String path;
   private volatile UUID uuid;
   private volatile AsyncExecutionStatus status = AsyncExecutionStatus.QUEUED;
   private volatile int progress = 0;
+  private volatile int page = 0;
```

Comment:
I didnt catch that the last time - but when you use synchronized, you dont need volatile. Synchronize blocks will already guarantee correctness. So its either volatile, or synchronized, but never both.